lib/Sabre/Location/LocationRoot.php

```diff
@@ -0,0 +1,155 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Photos\Sabre\Location;
+
+use OCA\Photos\DB\Location\LocationFile;
+use OCA\Photos\DB\Location\LocationInfo;
+use OCA\Photos\DB\Location\LocationMapper;
+use OCA\Photos\Service\ReverseGeoCoderService;
+use OCP\Files\IRootFolder;
+use Sabre\DAV\Exception\Forbidden;
+use Sabre\DAV\Exception\NotFound;
+use Sabre\DAV\ICollection;
+
+class LocationRoot implements ICollection {
+	protected LocationMapper $locationMapper;
+	protected ReverseGeoCoderService $reverseGeoCoderService;
+	protected LocationInfo $locationInfo;
+	protected IRootFolder $rootFolder;
+	/** @var LocationFile[] */
+	protected ?array $children = null;
+
+	public function __construct(
+		LocationMapper $locationMapper,
+		ReverseGeoCoderService $reverseGeoCoderService,
+		LocationInfo $locationInfo,
+		IRootFolder $rootFolder
+	) {
+		$this->locationMapper = $locationMapper;
+		$this->reverseGeoCoderService = $reverseGeoCoderService;
+		$this->locationInfo = $locationInfo;
+		$this->rootFolder = $rootFolder;
+	}
+
+	/**
+	 * @return never
+	 */
+	public function delete() {
+		throw new Forbidden('Not allowed to delete a location collection');
+	}
+
+	public function getName(): string {
+		return $this->reverseGeoCoderService->getLocationNameForLocationId(
+			$this->locationInfo->getLocationId()
+		);
+	}
+
+	/**
+	 * @return never
+	 */
+	public function setName($name) {
+		throw new Forbidden('Cannot change the location collection name');
+	}
+
+	/**
+	 * @param string $name
+	 * @param null|resource|string $data
+	 * @return never
+	 */
+	public function createFile($name, $data = null) {
+		throw new Forbidden('Cannot create a file in a location collection');
+	}
+
+	/**
+	 * @return never
+	 */
+	public function createDirectory($name) {
+		throw new Forbidden('Not allowed to create directories in this folder');
+	}
+
+	/**
+	 * @return LocationPhoto[]
+	 */
+	public function getChildren(): array {
+		if ($this->children === null) {
+			$this->children = array_map(
+				fn (LocationFile $file) => new LocationPhoto($this->locationInfo, $file, $this->rootFolder),
+				$this->locationMapper->findFilesForUserAndLocation($this->locationInfo->getUserId(), $this->locationInfo->getLocationId())
+			);
+		}
+
+		return $this->children;
+	}
+
+	public function getChild($name): LocationPhoto {
+		foreach ($this->getChildren() as $child) {
+			if ($child->getName() === $name) {
+				return $child;
+			}
+		}
+
+		throw new NotFound("$name not found");
+	}
+
+	public function childExists($name): bool {
+		try {
+			$this->getChild($name);
+			return true;
+		} catch (NotFound $e) {
+			return false;
+		}
+	}
+
+	public function getLastModified(): int {
+		return 0;
+	}
+
+	public function getFirstPhoto(): int {
+		return $this->getChildren()[0]->getFileId();
```

Comment:
What if there are no children?

Reply:
Not possible as the location would not exist, but I added a check just in case.